i18n/zh/docusaurus-plugin-content-docs/current/userguide/service/working-with-eriecanal.md

```diff
@@ -0,0 +1,415 @@
+---
+title: 与 ErieCanal 集成支持跨集群的服务发现和故障转移
+---
+
+这篇文档将演示如何使用 [ErieCanal](https://github.com/flomesh-io/ErieCanal) 完成 Karmada 多集群的互联互通，实现应用粒度的跨集群 Failover。
+
+ErieCanal 是一个 MCS（多集群服务 [Multi-Cluster Service](https://github.com/kubernetes-sigs/mcs-api)）实现，为 Kubernetes 集群提供 MCS、Ingress、Egress、GatewayAPI。
+
+## 总体架构
+
+在本示例中，我们借助 Karmada 实现资源的跨集群调度；借助 ErieCanal 实现服务的跨集群的注册发现。
+
+- 服务注册：通过创建 [`ServiceExport`](https://github.com/flomesh-io/ErieCanal/blob/7fc7e33315347ec69dc60ff19fdeb1cd1552ef34/apis/serviceexport/v1alpha1/serviceexport_types.go#L125) 资源来将 Service 声明为多集群服务，注册到 ErieCanal 的控制面，同时为该服务创建入口（Ingress）规则。
+- 服务发现：ErieCanal 的控制面，使用服务的信息及所在集群的信息创建 [`ServiceImport`](https://github.com/flomesh-io/ErieCanal/blob/7fc7e33315347ec69dc60ff19fdeb1cd1552ef34/apis/serviceimport/v1alpha1/serviceimport_types.go#L160) 资源，并将其同步到所有的成员集群。
+
+ErieCanal 会运行在控制面集群和成员集群上，接受集群注册的成为 ErieCanal 的控制面集群。ErieCanal 是独立的组件，无需运行在 Karmada 控制面上。前者负责多集群服务的注册发现，后者负责多集群的资源调度。
+
+![](../../resources/userguide/service/eriecanal/karmada-working-with-eriecanal-overview.png)
+
+完成服务的跨集群注册发现之后，在处理应用的访问（curl -> httpbin）时要完成流量的自动调度，这里有两种方案：
+
+- 集成服务网格 [osm-edge](https://flomesh.io/osm-edge/) 根据策略实现流量的调度，除了从 Kubernetes Service 获取服务的访问信息外，还会同多集群资源的 ServiceImport 中获取服务信息。
+- 使用 ErieCanal 中的组件 ErieCanalNet（该特性即将发布），通过 eBPF+sidecar（Node level）来实现流量的跨集群管理。
+
+下面是演示的全部流程，大家也可以使用我们提供的 [脚本 flomesh.sh](../../resources/userguide/service/eriecanal/flomesh.sh) 完成自动化地演示。使用脚本的前提，**系统需要已经安装 Docker 和 kubectl，并至少 8G 内存**。脚本的使用方式：
+
+- `flomesh.sh` - 不提供任何参数，脚本会创建 4 个集群、完成环境搭建（安装 Karmada、ErieCanal、osm-edge），并运行示例。
+- `flomesh.sh -h` - 查看参数的说明
+- `flomesh.sh -i` - 创建集群和完成搭建环境
+- `flomesh.sh -d` - 运行示例
+- `flomesh.sh -r` - 删除示例所在的命名空间
+- `flomesh.sh -u` - 销毁集群
+
+以下是分步指南，与 flomesh.sh 脚本的执行步骤一致。
+
+## 前提条件
+
+- 4 个集群：control-plane、clutser-1、cluster-2、cluster-3
+- Docker
+- k3d
+- helm
+- kubectl
+
+## 环境设置
+
+### 1. 安装 Karmada 控制面
+
+参考 [Karmada文档](https://karmada.io/zh/docs/installation/)，安装 Karmada 的控制面。Karmada 初始化完成后，将 **三个成员集群** clutser-1、cluster-2、cluster-3 注册到 Karmada 控制面，可 [参考 Karmada 的集群注册指引](https://karmada.io/zh/docs/userguide/clustermanager/cluster-registration/)。
+
+这里使用 push 模式，集群注册参考下面的命令（在控制面集群 control-plane 上执行 ）：
+
+```shell
+karmadactl --kubeconfig PATH_TO_KARMADA_CONFIG join CLUSTER_NAME --cluster-kubeconfig=PATH_CLSUTER_KUBECONFIG
+```
+
+接来下，需要向 Karmada 控制面注册 ErieCanal 的多集群 CRD。
+
+```shell
+kubectl --kubeconfig PATH_TO_KARMADA_CONFIG apply -f https://raw.githubusercontent.com/flomesh-io/ErieCanal/main/charts/erie-canal/apis/flomesh.io_clusters.yaml
```

Comment:
It seems that you install cluster CRD of flomesh in the karmada control plane. But Karmada also has `cluster` apiservice. I wonder if there is a conflict when users use `kubectl get cluster`.

Reply:
Clear with your concert. But one cluster CRD provided by Karmada exists in Karmada apiserver only, while another exists in Kubernetes apiserver. What's why we got Karmada cluster when specifying Karmada apiserver, and got ErieCanal cluster in Kubernetes one.

```shell
kubectl --kubeconfig PATH_TO_KARMADA_CONFIG get cluster
NAME        VERSION        MODE   READY   AGE
cluster-1   v1.23.8+k3s2   Push   True    154m
cluster-2   v1.23.8+k3s2   Push   True    154m
cluster-3   v1.23.8+k3s2   Push   True    154m

kubectl get cluster
NAME        REGION    ZONE      GROUP     GATEWAY HOST   GATEWAY PORT   MANAGED   MANAGED AGE   AGE
local       default   default   default                  80                                     159m
cluster-1   default   default   default   10.0.2.4       80             True      159m          159m
cluster-2   default   default   default   10.0.2.5       80             True      159m          159m
cluster-3   default   default   default   10.0.2.6       80             True      159m          159m
```

Reply:
Got it.

Reply:
But I find that we also install flomesh_cluster api in karmada control plane. 
```
kubectl --kubeconfig PATH_TO_KARMADA_CONFIG apply -f https://raw.githubusercontent.com/flomesh-io/ErieCanal/main/charts/erie-canal/apis/flomesh.io_clusters.yaml
```